autograder/question.py

```diff
@@ -169,13 +181,21 @@ class AutograderFailError(RuntimeError):
 
     pass
 
+class AutograderHardFailError(RuntimeError):
+    """
+    This error indicates that hard_fail() has been called on a question
+    and execution should be stopped for all questions in the assignment.
+    """
+
+    pass
+
 class GradedQuestion(object):
```

Comment:
to_dict() is missing hard_fail and skipped.
Note that this is pretty important, this means that this data would never be available to the autograder.